### PR TITLE
Use new crd-schema-gen tool

### PR DESCRIPTION
**Pull request by damemi**
This PR:
1. Updates Makefile to use new, updated, containerized crd-schema-gen
2. Removes old deps based on forks of upstream deps for the old crd-gen scripts
3. Bumps deps for API changes and to clear out deps that were just removed
4. Updates CRD to pass verify

### CODE REVIEW

**Comment by damemi** [comment]
/hold
seeing how this plays out in other repos before I start applying this change everywhere, but I still want to open it now for review

File: Makefile [hunk 1]
```diff
@@ -44,11 +44,13 @@ clean:
...
 .PHONY: clean
 
+PKG := ${PWD:${GOPATH}/%=%}
 update-codegen-crds:
-	go run ./vendor/github.com/openshift/library-go/cmd/crd-schema-gen/main.go --domain openshift.io --apis-dir vendor/github.com/openshift/api
+	docker run -v ${PWD}:/go/$(PKG):Z -w /go/$(PKG) registry.svc.ci.openshift.org/ocp/4.2:crd-schema-gen --apis-dir vendor/github.com/openshift/api/operator/v1
 update-codegen: update-codegen-crds
 verify-codegen-crds:
-	go run ./vendor/github.com/openshift/library-go/cmd/crd-schema-gen/main.go --domain openshift.io --apis-dir vendor/github.com/openshift/api --verify-only
+	go get github.com/openshift/crd-schema-gen/... && go install github.com/openshift/crd-schema-gen/...
```
Comment (sttts) [inline Makefile]: this must be versioned

**Comment by adambkaplan** [comment]
@damemi you'll want to rebase and re-work bump(*) commits, as we're now using build-machinery in library-go

**Comment by damemi** [comment]
@adambkaplan pushed some updates, this is the method we decided on to use in KAS-O, hopefully it passes CI here too

**Comment by adambkaplan** [comment]
@damemi has the same tooling been added to our 4.1.z-stream branches?

**Pull request merged**